crates/router/src/core/payments/helpers.rs

```diff
@@ -285,16 +288,53 @@ pub async fn get_token_pm_type_mandate_details(
     Option<String>,
     Option<storage_enums::PaymentMethod>,
     Option<storage_enums::PaymentMethodType>,
-    Option<api::MandateData>,
+    Option<MandateData>,
     Option<payments::RecurringMandatePaymentData>,
     Option<String>,
 )> {
+    let mandate_data = request.mandate_data.clone().map(|d| MandateData {
+        customer_acceptance: d.customer_acceptance.map(|d| {
+            data_models::mandates::CustomerAcceptance {
+                acceptance_type: match d.acceptance_type {
+                    api::AcceptanceType::Online => data_models::mandates::AcceptanceType::Online,
+                    api::AcceptanceType::Offline => data_models::mandates::AcceptanceType::Offline,
+                },
```

Comment:
Better to define this in a conversion trait

Reply:
Done!!!